libindi/libs/indicom.c

```diff
@@ -347,6 +354,22 @@ int tty_write(int fd, const char *buf, int nbytes, int *nbytes_written)
 #ifdef _WIN32
     return TTY_ERRNO;
 #else
+    int buffer[66]={0};
+    char *byteBuffer = (char *)buffer;
+
+    if (ttyGeminiUdpFormat)
+    {
+        buffer[0] = ++sequenceNumber;
+        buffer[1] = 0;
+        strncpy((char *)&buffer[2], buf, nbytes);
+        // Add on the 8 bytes for the header and 1 byte for the null terminator
+        nbytes += 9;
+    }
+    else
+    {
+        strncpy(byteBuffer, buf, 255);
```

Comment:
This line broke my dome driver as strncpy ends at the first zero byte and it's not using a text based protocol but purely binary. Should be memcpy or the whole copy in itself is totally unnecessary in the non-gemini case.